src/common/slash.ts

```diff
@@ -0,0 +1,20 @@
+import { ApplicationCommandOptionData, CommandInteraction } from 'discord.js';
+import { IContainer } from './types';
+
+export const slashCommands: Record<string, ISlashCommand> = {};
+
+export interface ISlashCommand {
+  commandName: string;
+  name: string;
+  description: string;
+  parameters?: ApplicationCommandOptionData[];
+  execute({
+    interaction,
+    container,
+  }: {
+    interaction: CommandInteraction;
+    container: IContainer;
+  }): void | Promise<void>;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  [key: string]: any;
```

Comment:
Is there a specific reason this is needed?

Reply:
I now understand the issue you're pointing out, however, I don't think this is an optimal solution. Using `[key: string]: any` causes the return type of methods within an object to always return `any`. For example, in the 8Ball plugin `this._createEmbed()` returns `any` despite being typed to return `MessageEmbed`.

Alternatively, there's two things we can do to avoid this:

- Continue to use plain objects, and have a separate export statement:
```ts
const replyOptions = [...];

const 8ballPlugin: ISlashCommand = {
	name: 'Magic Eight Ball',
	commandName: '8ball',
	...
	execute({ interaction }) {
    	interaction.reply({ embeds: [createEmbed()] });
  	},
};

function createEmbed() {
  return new MessageEmbed()
      .setTitle(plugin.name)
      .setThumbnail('https://publicdomainvectors.org/photos/Jarno_8_ball.png')
      .setDescription(getRandom(replyOptions));
}

export default 8ballPlugin;
```

- Use classes with `implements`
```ts
export default class Plugin implements ISlashCommand {
  commandName = 'Magic Eight Ball';
  name = '8ball';
  description = 'Get answers from a pseudorandom number generator.';
  execute({ interaction }: { interaction: CommandInteraction<CacheType>; container: IContainer; }): void | Promise<void> {
    interaction.reply({ embeds: [createEmbed()] });
  }

  createEmbed() {
    return new MessageEmbed()
        .setTitle(this.name)
        .setThumbnail('https://publicdomainvectors.org/photos/Jarno_8_ball.png')
        .setDescription(getRandom(replyOptions));
  }
}
```

I don't have a strong opinion on either of these, but both are more type-safe than the current approach.

Reply:
I believe the class error is due to the method params when using `implements` need explicit types.